lib/response.js

```diff
@@ -229,7 +237,31 @@ res.send = function send(body) {
     this.end();
   } else {
     // respond
-    this.end(chunk, encoding);
+    if (isChunkBlob) {
+      /**
+       * @type {http.ServerResponse}
+       */
+      var res = this;
+      /**
+       * @type {Blob}
+       */
+      var blob = chunk;
+
+      if(typeof blob.stream === 'function') {
+        var inputStream = Readable.from(blob.stream());
+        inputStream.on('error', req.next)
+        res.on('error', req.next)
```

Comment:
You probably need to remove this handlers before proceeding. The error handler itself that `next` may try to write to `res` and cause an error, and this would end up involing `next` a second time.

Reply:
Also on this res error, it probably means we need to stop the pipe from the blob so it doesn't keep writing to res.